docs: rekeying transaction (#582)
scale-it/algo-builder: docs/guide/runtime-advanced.md

---
layout: splash
---

# Rekeying transaction

- Read first about [Algorand rekeying](https://developer.algorand.org/docs/get-details/accounts/rekey/). 
`Runtime` supports a rekey transaction. You can use field `rekeyTo` in `payFlags` to rekey:
- Account to any account.
- Smart Signature account to any account.
- Application to any account. We use inner transaction to rekey application.

Example: 

> We can rekey John account to Bob follow suing the following transaction params.

Reviewer: ```suggestion
We can rekey John account to Bob using the following transaction params.
```

---
layout: splash
---

# Rekeying transaction

> - You should read about Algorand rekeying, [here](https://developer.algorand.org/docs/get-details/accounts/rekey/). 

Reviewer 2: ```suggestion
Read first about [Algorand rekeying](https://developer.algorand.org/docs/get-details/accounts/rekey/). 
```

> - `Runtime` supported a rekeying transaction. You can use field `rekeyTo` in `payFlags` to rekey from account to another address. 
> - We can rekeying:
>     - Account to any account.
>     - Logic account to any account.
>     - Application to any account. We use inner transaction to rekey application.

Reviewer 2: ```suggestion
`Runtime` supports a rekey transaction. You can use field `rekeyTo` in `payFlags` to rekey:
- Account to any account.
- Smart Signature account to any account.
- Application to any account. We use inner transaction to rekey application.
```

Example: 

> - We can rekey join account to bob follow below config.

Reviewer 2: ```suggestion
We can rekey John account to Bob follow suing the following transaction parms.
```

```js
    type: TransactionType.TransferAlgo,
    sign: SignType.SecretKey,
    fromAccount: john,
    toAccountAddr: alice.address,
    amountMicroAlgos: 0n,
    payFlags: { 
        totalFee: fee 
        rekeyTo: bob.address
    } 
```

> - We can rekey logic signature (lsig) to bob.

Reviewer 2: ```suggestion
We can rekey smart signature (lsig) to bob.
```

```js
{
    type: types.TransactionType.TransferAlgo,
    sign: types.SignType.LogicSignature,
    lsig: lsig,
    toAccountAddr: bob.address,
    amountMicroAlgos: 0n,
    payFlags: {
        totalFee: fee 
        rekeyTo: bob.address 
    }
}
```

> You can check how to use rekeying transaction at  [here](https://github.com/scale-it/algo-builder/blob/master/packages/runtime/test/integration/rekey-transaction.ts).

Reviewer 2: ```suggestion
You can check how to use rekeying transaction in the [source code](https://github.com/scale-it/algo-builder/blob/master/packages/runtime/test/integration/rekey-transaction.ts).
```
